.github/workflows/publish-on-pypi.yml

```diff
@@ -58,7 +90,7 @@ jobs:
   docs:
     runs-on: ubuntu-latest
     needs:
-    - deploy
+    - publish
```

Comment:
A missed needs or an intentional needs?

Reply:
Intentional to have a flow for the publish process. I want to make sure the package gets deployed to PyPi before building docs.